src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Private/JavaScriptMarshalImpl.JsInvoke.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#nullable disable
+
+namespace System.Runtime.InteropServices.JavaScript.Private
+{
+    internal static unsafe partial class JavaScriptMarshalImpl
+    {
+        internal static unsafe void InvokeJSFunctionByName<T1>(string functionName, T1 arg1)
+        {
+            var signature = GetMethodSignature(null, typeof(T1), typeof(void));
+            var excMessage = default(string);
+            var buffer = stackalloc byte[signature.TotalBufferLength];
+            var args = JavaScriptMarshal.CreateArguments(buffer);
+            JavaScriptMarshal.InitVoid(ref excMessage, args);
+            JavaScriptMarshal.InitArgument(1, ref arg1, args, signature);
+            ToJavaScriptDynamic(ref arg1, args[1]);
+
+            _InvokeJSFunction1(string.Intern(functionName), signature.Header, args.Buffer);
```

Comment:
`string.Intern` keeps the string alive indefinitely. It will create a leak as different strings are passed around.

I think this needs to be a cache that remembers e.g. last N strings. There are number of caches like that floating around. For example: https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/Common/TypeSystem/Ecma/CachingMetadataStringDecoder.cs

Reply:
The intended use case is that they're compile time string literals, not for the caller to actually use string.Intern. At the point where we're maintaining a bunch of custom string caches on the JS and C# side we're adding a bunch of additional problems when the number of JS targets is going to be in the dozens. (If the end user wants to call JS methods dynamically, they can use a custom JS function to do that.)

Reply:
I think we could stop caching the names and also JS function on JS side for this scenario. Because this is dynamic and slow way, the fast and generated way doesn't have this need either. Also, it's not clear that this dynamic invoke would be included in the final PR, but it made prototyping easier.

Reply:
I removed caching and string interning